### ISSUE TITLE
Can't run chart falco-1.18.5 via Helm in OpenSUSE nodes

### ISSUE THREAD

**Issue by mattups** [falcosecurity/falco#2029]
**Describe the bug**
Running helm chart `falco-1.18.5` on Kubernetes OpenSUSE nodes is not possibile as far as it seems due to broken module download.

After installing `falco` via Helm, falco containers are running only on Ubuntu nodes.
These are the logs from a falco container running on a OpenSUSE node:
```
* Setting up /usr/src links from host
* Running falco-driver-loader for: falco version=0.31.1, driver version=b7eb0dd65226a8dc254d228c8d950d07bf3521d2
* Running falco-driver-loader with: driver=module, compile=yes, download=yes
* Unloading falco module, if present
* Looking for a falco module locally (kernel 5.3.18-lp152.106-default)
* Trying to download a prebuilt falco module from https://download.falco.org/driver/b7eb0dd65226a8dc254d228c8d950d07bf3521d2/falco_opensuse-leap_5.3.18-lp152.106-default_1.ko
curl: (22) The requested URL returned error: 404
Unable to find a prebuilt falco module
* Trying to dkms install falco module with GCC /usr/bin/gcc
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
* Running dkms build failed, couldn't find /var/lib/dkms/falco/b7eb0dd65226a8dc254d228c8d950d07bf3521d2/build/make.log (with GCC /usr/bin/gcc)
* Trying to dkms install falco module with GCC /usr/bin/gcc-8
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
* Running dkms build failed, couldn't find /var/lib/dkms/falco/b7eb0dd65226a8dc254d228c8d950d07bf3521d2/build/make.log (with GCC /usr/bin/gcc-8)
* Trying to dkms install falco module with GCC /usr/bin/gcc-6
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
* Running dkms build failed, couldn't find /var/lib/dkms/falco/b7eb0dd65226a8dc254d228c8d950d07bf3521d2/build/make.log (with GCC /usr/bin/gcc-6)
* Trying to dkms install falco module with GCC /usr/bin/gcc-5
DIRECTIVE: MAKE="'/tmp/falco-dkms-make'"
* Running dkms build failed, couldn't find /var/lib/dkms/falco/b7eb0dd65226a8dc254d228c8d950d07bf3521d2/build/make.log (with GCC /usr/bin/gcc-5)
* Trying to load a system falco module, if present
Consider compiling your own falco driver and loading it or getting in touch with the Falco community
Tue May 31 12:27:10 2022: Falco version 0.31.1 (driver version b7eb0dd65226a8dc254d228c8d950d07bf3521d2)
Tue May 31 12:27:10 2022: Falco initialized with configuration file /etc/falco/falco.yaml
Tue May 31 12:27:10 2022: Loading rules from file /etc/falco/falco_rules.yaml:
Tue May 31 12:27:11 2022: Loading rules from file /etc/falco/falco_rules.local.yaml:
Tue May 31 12:27:11 2022: Unable to load the driver.
Tue May 31 12:27:11 2022: Runtime error: error opening device /host/dev/falco0. Make sure you have root credentials and that the falco module is loaded.. Exiting.
```

After looking at this I tried to go on the nodes and try to install it manually with the [official docs](https://falco.org/docs/getting-started/installation/#suse):
```
~ # rpm --import https://falco.org/repo/falcosecurity-3672BA8F.asc
~ # curl -s -o /etc/zypp/repos.d/falcosecurity.repo https://falco.org/repo/falcosecurity-rpm.repo
~ # zypper -n install kernel-default-devel-$(uname -r | sed s/\-default//g)
Building repository 'falcosecurity-rpm' cache ............................................................................................................................................................................[done]
Retrieving repository 'Update repository of openSUSE Backports' metadata .................................................................................................................................................[done]
Building repository 'Update repository of openSUSE Backports' cache ......................................................................................................................................................[done]
Retrieving repository 'Update repository with updates from SUSE Linux Enterprise 15' metadata ............................................................................................................................[done]
Building repository 'Update repository with updates from SUSE Linux Enterprise 15' cache .................................................................................................................................[done]
Retrieving repository 'openSUSE-Backports-15.3-Updates for x86_64' metadata ..............................................................................................................................................[done]
Building repository 'openSUSE-Backports-15.3-Updates for x86_64' cache ...................................................................................................................................................[done]
Retrieving repository 'openSUSE-Leap-15.3-Updates for x86_64' metadata ...................................................................................................................................................[done]
Building repository 'openSUSE-Leap-15.3-Updates for x86_64' cache ........................................................................................................................................................[done]
Retrieving repository 'SLE-15-SP3-Updates for x86_64 openSUSE 15.3' metadata .............................................................................................................................................[done]
Building repository 'SLE-15-SP3-Updates for x86_64 openSUSE 15.3' cache ..................................................................................................................................................[done]
Loading repository data...
Reading installed packages...
Package 'kernel-default-devel-5.3.18-150300.59.54' not found.
```

**How to reproduce it**
Install falco chart via Helm.
```
helm repo add falcosecurity https://falcosecurity.github.io/charts
helm repo update
helm install falco falcosecurity/falco -f values.yaml
```

**Expected behaviour**
Falco containers running on both Ubuntu and OpenSUSE nodes

**Environment**
- Falco version (from a container running on Ubuntu):
```
:/# falco --version
Falco version: 0.31.1
Driver version: b7eb0dd65226a8dc254d228c8d950d07bf3521d2
```
Helm chart version: 1.18.5
Kubernetes cluster 1.20.11 (rke) with mixed nodes based on:
```
> kubectl get nodes -o wide
ROLES                      AGE     VERSION   OS-IMAGE                              KERNEL-VERSION                CONTAINER-RUNTIME
controlplane,etcd,worker   2y88d   v1.20.11  SUSE Linux Enterprise Server 15 SP3   5.3.18-150300.59.60-default   docker://20.10.12-ce
controlplane,etcd,worker   624d    v1.20.11  SUSE Linux Enterprise Server 15 SP3   5.3.18-150300.59.49-default   docker://20.10.12-ce
controlplane,etcd,worker   2y88d   v1.20.11  SUSE Linux Enterprise Server 15 SP2   5.3.18-24.96-default          docker://20.10.9-ce
worker                     377d    v1.20.11  openSUSE Leap 15.2                    5.3.18-lp152.106-default      docker://20.10.9-ce
worker                     499d    v1.20.11  openSUSE Leap 15.2                    5.3.18-lp152.106-default      docker://20.10.9-ce
worker                     364d    v1.20.11  openSUSE Leap 15.3                    5.3.18-150300.59.54-default   docker://20.10.12-ce
worker                     378d    v1.20.11  openSUSE Leap 15.2                    5.3.18-lp152.106-default      docker://20.10.9-ce
worker                     249d    v1.20.11  Ubuntu 20.04.4 LTS                    5.4.0-91-generic              docker://20.10.7
worker                     249d    v1.20.11  Ubuntu 20.04.4 LTS                    5.4.0-86-generic              docker://20.10.7
worker                     657d    v1.20.11  openSUSE Leap 15.2                    5.3.18-lp152.106-default      docker://20.10.9-ce
worker                     657d    v1.20.11  openSUSE Leap 15.3                    5.3.18-150300.59.49-default   docker://20.10.12-ce`
````
- System info: Can't retrieve it from an OpenSUSE node.
- Cloud provider or hardware configuration: Rancher RKE, vSphere
- OS: See `kubectl get nodes...` above
- Kernel: See `kubectl get nodes...` above
- Installation method:
Kubernetes, Helm

**Comment by alacuku** [comment]
Hi @mattups, as far as i know there are not prebuild kernel driver modules for OpenSUSE that's why [falco driver loader](https://github.com/falcosecurity/falco/blob/master/scripts/falco-driver-loader) fails to download the module. First thing the `falco driver loader` checks for the existence of local driver module and loads it if found. Otherwise it tries to download a prebuild driver module from [here](https://download.falco.org/?prefix=driver/) based on the kernel version of the machine where it is running. If the prebuild driver module is not available then it tries to build it from source, failing if any of Falco dependencies or kernel headers are missing.

A viable alternative to get Falco running in your nodes could be the following:

1. install Falco [dependencies](https://falco.org/docs/getting-started/source/) and kernel headers on each node;
2. install again Falco using the helm chart or just restart the Falco pods on the affected nodes

Now the `falco driver loader` should build the driver module from source.

**Comment by mattups** [comment]
Tried also with `bpf` since `driver` wasn't loading, but no luck so far.
My steps taken from the [docs](https://falco.org/docs/getting-started/source/):
(tried with both `master` branch and `0.32.0` tag
```shell
zypper -n install clang llvm
cmake -DBUILD_BPF=ON ..
make bpf

:~/falco/build # make bpf
expr: syntax error: unexpected argument '1'
expr: syntax error: unexpected argument '1'
In file included from /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.c:13:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/sched.h:14:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/pid.h:6:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/wait.h:9:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/spinlock.h:89:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/spinlock.h:27:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/qspinlock.h:110:
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/qspinlock.h:67:55: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
        return likely(atomic_try_cmpxchg_acquire(&lock->val, &val, _Q_LOCKED_VAL));
                                                             ^~~~
/usr/src/linux-5.3.18-150300.59.68/include/linux/compiler.h:77:40: note: expanded from macro 'likely'
# define likely(x)      __builtin_expect(!!(x), 1)
                                            ^
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/atomic-instrumented.h:691:38: note: passing argument to parameter 'old' here
atomic_try_cmpxchg(atomic_t *v, int *old, int new)
                                     ^
In file included from /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.c:13:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/sched.h:14:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/pid.h:6:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/wait.h:9:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/spinlock.h:89:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/spinlock.h:27:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/qspinlock.h:110:
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/qspinlock.h:81:52: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
        if (likely(atomic_try_cmpxchg_acquire(&lock->val, &val, _Q_LOCKED_VAL)))
                                                          ^~~~
/usr/src/linux-5.3.18-150300.59.68/include/linux/compiler.h:77:40: note: expanded from macro 'likely'
# define likely(x)      __builtin_expect(!!(x), 1)
                                            ^
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/atomic-instrumented.h:691:38: note: passing argument to parameter 'old' here
atomic_try_cmpxchg(atomic_t *v, int *old, int new)
                                     ^
In file included from /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.c:13:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/sched.h:14:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/pid.h:6:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/wait.h:9:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/spinlock.h:89:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/spinlock.h:43:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/qrwlock.h:6:
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/qrwlock.h:65:56: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
        return likely(atomic_try_cmpxchg_acquire(&lock->cnts, &cnts,
                                                              ^~~~~
/usr/src/linux-5.3.18-150300.59.68/include/linux/compiler.h:77:40: note: expanded from macro 'likely'
# define likely(x)      __builtin_expect(!!(x), 1)
                                            ^
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/atomic-instrumented.h:691:38: note: passing argument to parameter 'old' here
atomic_try_cmpxchg(atomic_t *v, int *old, int new)
                                     ^
In file included from /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.c:13:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/sched.h:14:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/pid.h:6:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/wait.h:9:
In file included from /usr/src/linux-5.3.18-150300.59.68/include/linux/spinlock.h:89:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/spinlock.h:43:
In file included from /usr/src/linux-5.3.18-150300.59.68/arch/x86/include/asm/qrwlock.h:6:
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/qrwlock.h:92:53: warning: passing 'u32 *' (aka 'unsigned int *') to parameter of type 'int *' converts between pointers to integer types with different sign [-Wpointer-sign]
        if (likely(atomic_try_cmpxchg_acquire(&lock->cnts, &cnts, _QW_LOCKED)))
                                                           ^~~~~
/usr/src/linux-5.3.18-150300.59.68/include/linux/compiler.h:77:40: note: expanded from macro 'likely'
# define likely(x)      __builtin_expect(!!(x), 1)
                                            ^
/usr/src/linux-5.3.18-150300.59.68/include/asm-generic/atomic-instrumented.h:691:38: note: passing argument to parameter 'old' here
atomic_try_cmpxchg(atomic_t *v, int *old, int new)
                                     ^
In file included from /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.c:23:
/root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/fillers.h:3147:49: error: no member named 'features' in 'struct io_uring_params'
        features = io_uring_setup_feats_to_scap(params.features);
                                                ~~~~~~ ^
4 warnings and 1 error generated.
make[7]: *** [/root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/Makefile:53: /root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf/probe.o] Error 1
make[6]: *** [/usr/src/linux-5.3.18-150300.59.68/Makefile:1675: _module_/root/falco/build/falcosecurity-libs-repo/falcosecurity-libs-prefix/src/falcosecurity-libs/driver/bpf] Error 2
make[5]: *** [../../../linux-5.3.18-150300.59.68/Makefile:179: sub-make] Error 2
make[4]: *** [Makefile:38: all] Error 2
make[3]: *** [driver/bpf/CMakeFiles/bpf.dir/build.make:74: driver/bpf/CMakeFiles/bpf] Error 2
make[2]: *** [CMakeFiles/Makefile2:1007: driver/bpf/CMakeFiles/bpf.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:1014: driver/bpf/CMakeFiles/bpf.dir/rule] Error 2
make: *** [Makefile:398: bpf] Error 2
```

**Comment by alacuku** [comment]
Hi @mattups, thanks for the info. I'm gonna give a try and reproduce the issue. I will get back to you as soon as possible!

**Comment by alacuku** [comment]
Hi @mattups, the compile error you are getting while building the `bpf probe` is now fixed, please see falcosecurity/libs#379.

Could you please try to build the `bpf probe` using `falco 0.31.1`?

**Comment by alacuku** [comment]
Furthermore, the error you are receiving when loading the kernel module:

`:~/falco/build # sudo insmod driver/falco.ko
insmod: ERROR: could not insert module driver/falco.ko: Key was rejected by service`

could be related to `secure boot` that requires the module to be `signed `in order to load it.

Could you check if `secure boot` is enabled for those `vms`?
